### ISSUE TITLE
Wrong labels fetched in CV folds when test fraction >0

### ISSUE THREAD

**Issue by jsk389** [tasoc/starclass#33]
When running cross-validation and setting the tf flag to anything greater than zero the wrong training and test labels are fetched. The cross-validation does not play nicely when a holdout test set is needed outside of the cross-validation.

I wonder if it comes from the line below, because the class is being copied with tf > 0 and so it goes through the train_test_split again which it shouldn't do. This is because the tf in the __init__ of the TrainingSet class refers to the test fraction argument, but when it's being reinitialised below it is referring to the cross-validation fold. This is of course doesn't matter when tf=0, but for tf>0 this causes the above problems. 

https://github.com/tasoc/starclass/blob/f25836f96ef590d342a0bda2e94756fa8f52007e/starclass/training_sets/training_set.py#L72

I unfortunately have to idea how to solve this, but without this we can't properly run the meta-classifier with a holdout set that hasn't been involved in the training of any of the classifiers.

**Comment by jsk389** [comment]
This has been fixed. It turns out it's due to the fact that `train_test_split` shuffles the indices and so this apparently causes problems when fetching the labels. I will push the fix later today.
